integration-test/src/main/java/com/sequenceiq/it/cloudbreak/actor/CloudbreakUserCache.java

```diff
@@ -1,66 +1,152 @@
 package com.sequenceiq.it.cloudbreak.actor;
 
-import java.io.IOException;
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
 
 import com.fasterxml.jackson.core.type.TypeReference;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
 import com.sequenceiq.cloudbreak.common.json.JsonUtil;
 import com.sequenceiq.cloudbreak.util.FileReaderUtils;
+import com.sequenceiq.it.cloudbreak.cloud.v4.CommonCloudProperties;
 import com.sequenceiq.it.cloudbreak.exception.TestFailException;
 
+@Component
 public class CloudbreakUserCache {
 
-    private static volatile CloudbreakUserCache instance;
-
-    private static Object mutex = new Object();
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudbreakUserCache.class);
 
     private Map<String, List<CloudbreakUser>> usersByAccount;
 
-    private CloudbreakUserCache() {
+    private Map<String, String> initializedKeys;
+
+    private Set<String> accountIds;
+
+    @Value("${integrationtest.user.realUms.accountKey:default}")
+    private String realUmsUserAccount;
+
+    @Value("${integrationtest.user.realUms.environmentKey:dev}")
+    private String realUmsUserEnvironment;
+
+    @Inject
+    private CommonCloudProperties commonCloudProperties;
+
+    protected CommonCloudProperties commonCloudProperties() {
+        return commonCloudProperties;
     }
 
-    public static CloudbreakUserCache getInstance() {
-        CloudbreakUserCache result = instance;
-        if (result == null) {
-            synchronized (mutex) {
-                result = instance;
-                if (result == null) {
-                    result = new CloudbreakUserCache();
-                    instance = result;
-                }
-            }
-        }
-        return result;
+    public CloudbreakUser getByDisplayName(String name) {
+        return getByDisplayName(name, commonCloudProperties.getUser().getRealUms().getEnvironmentKey(),
+                commonCloudProperties.getUser().getRealUms().getAccountKey());
     }
 
-    public CloudbreakUser getByName(String name) {
-        if (usersByAccount == null) {
-            initUsers();
+    public CloudbreakUser getByDisplayName(String name, String environmentKey, String accountKey) {
+        setRealUmsUserEnvironment(environmentKey);
```

Comment:
Why do we need to set a `Compomnent`'s field here? Is this even thread safe? 
Also it's not a really good pattern to change a @Value annotation on the go. Can't this initialization be done in a PostConstruct? and only be done once? Since we agreed we only run tests in a single account.

Reply:
Actually we run tests in 2 separate accounts in parallel at [cloudbreak-real-ums-test](http://ci-cloudbreak.eng.hortonworks.com/job/cloudbreak-real-ums-test/) and also at [integration-test-pr-real-ums-test
](http://ci-cloudbreak.eng.hortonworks.com/job/integration-test-pr-real-ums-test/).
However very useful if we can use environment variables or application parameters to set up default values for these tests.
Furthermore as far as I know Legacy tests wont be with us after a while and we can step forward to single account for single run concept.

Reply:
Sounds good! Separate test jobs are sound feasible and more maintainable for me as well. I buy the idea!